### PR TITLE
Remove defaults in osx

### PR DESCRIPTION
**Pull request by isuruf**
conda-forge is mostly self sustaining if we bump to clang 8.0.1.

A few remaining packages
- ~~ld64~~
- ~~cctools~~
- mkl*

### CODE REVIEW

**Comment by scopatz** [comment]
This seems reasonable to me

**Comment by jakirkham** [comment]
Is using `defaults` causing us problems on macOS?

**Comment by CJ-Wright** [comment]
Is this still needed now that we are on 9? https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L14

**Comment by ocefpaf** [comment]
@isuruf should we refresh this one or close it?

**Comment by isuruf** [comment]
We should have a look at which packages are coming from defaults on osx. Anybody from @conda-forge/bot interested?

**Comment by beckermr** [comment]
I see very little in the envs on my mac right now. We shipped mkl a while back, so that can be crossed off the list.

**Comment by beckermr** [comment]
I don't have the time to do a proper census though.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ c_compiler_version:            # [unix]
   - 7                          # [linux64]
   - 7.5                        # [aarch64]
   - 8                          # [ppc64le]
-  - 8.3                        # [armv7l]
+  - 8.3                        # [armv7l
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:17]: typo?

**Pull request closed without merging**